### PR TITLE
topology2: fixup google aec topology

### PR DESCRIPTION
**Pull request by RanderWang**

### CODE REVIEW

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf
[the remark this answers is not in the document]
Comment (RanderWang) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: No, please check the pictures
correct
incorrect

[the remark this answers is not in the document]
Comment (lgirdwood) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: @RanderWang is top picture "before" and bottom picture "after" this PR ?

[the remark this answers is not in the document]
Comment (RanderWang) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: @lgirdwood  The bottom picture is generated before this PR which removes the link beween module-copier.18.2 and host-copier.27.capture

[the remark this answers is not in the document]
Comment (lgirdwood) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: @yongzhi1 good for you now ?

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf [hunk 1]
```diff
@@ -89,10 +89,6 @@ Object.Base.route [
 		source	$GOOGLE_RTC_AEC_REF_SOURCE
 		sink    "google-rtc-aec.18.1"
 	}
-	{
-		source	"module-copier.18.2"
-		sink    "host-copier.27.capture"
-	}
```
Comment (RDharageswari) [inline tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:95]: on main 29 is Echo reference and 27 is DMIC RTC AEC. So this doesnt work on "main"

**Review by RDharageswari — CHANGES_REQUESTED** [review]

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf (hunk 1 above)
Comment (RanderWang) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf]: thanks.  Echo reference was updated to 29. updated

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf
[the remark this answers is not in the document]
Comment (RanderWang) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: @lgirdwood  Yong is enjoying his vocation. @RDharageswari has validated this PR and gave me some feedback.

**Review by yongzhi1 — APPROVED** [review]

**Comment by lgirdwood** [comment]
@RDharageswari changes good for you ?

**Review by RDharageswari — APPROVED** [review]

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf
[the remark this answers is not in the document]
Comment (RDharageswari) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: i verified this PR.One question on this.
i see the following o/p:
rex-rev1 ~ # arecord -l
**** List of CAPTURE Hardware Devices ****
card 0: sofrt5682 [sof-rt5682], device 0: Headset (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofrt5682 [sof-rt5682], device 2: Bluetooth (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofrt5682 [sof-rt5682], device 27: DMIC0 RTC AEC (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofrt5682 [sof-rt5682], device 28: DMIC0 RTC AEC NS (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofrt5682 [sof-rt5682], device 29: EchoRef (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofrt5682 [sof-rt5682], device 99: DMIC Raw (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
rex-rev1 ~ #

This sequence doesnt work:
1)start speaker playback
2)start dmic capture:  arecord -Dhw:0,27 /tmp/tmp.wav -c4 -r48000 -fs32_le -vv

there is a strong dependency on echo reference endpoint to be started.
While this works:
1)start speaker playback
2)start echo ref capture
3)start dmic capture: arecord -Dhw:0,27 /tmp/tmp.wav -c4 -r48000 -fs32_le -vv.

There is no need for the echo reference endpoint to be opened from the user space, while speaker playback the data can be fed to AEC without the FE.

**Review by RDharageswari — COMMENTED** [review]
can you help on my question?

File: tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf
[the remark this answers is not in the document]
Comment (lgirdwood) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: @RDharageswari Can you log this as an issue if not already done so. Thanks !

[the remark this answers is not in the document]
Comment (RanderWang) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: The first design was to use 3 stream, 2 capture and one playback for speaker. Your requirement is a new feature. We need a loop back module in playback pipeline to dmic pipeline.

[the remark this answers is not in the document]
Comment (RDharageswari) [reply tools/topology/topology2/platform/intel/google-rtc-aec-reference.conf:1]: > @RDharageswari Can you log this as an issue if not already done so. Thanks !

Sure Liam!

**Pull request merged**